src/abstractdataframe/show.jl

```diff
@@ -2,32 +2,32 @@ Base.summary(df::AbstractDataFrame) = # -> String
     @sprintf("%d×%d %s", size(df)..., typeof(df).name)
 
 let
-    local io = IOBuffer(Vector{UInt8}(undef, 80), read=true, write=true)
+    local buffer = IOBuffer(Vector{UInt8}(undef, 80), read=true, write=true)
     global ourstrwidth
 
     """
-        DataFrames.ourstrwidth(x::Any)
+        DataFrames.ourstrwidth(io::IO, x::Any)
 
     Determine the number of characters that would be used to print a value.
     """
-    function ourstrwidth(x::Any) # -> Int
-        truncate(io, 0)
-        ourshowcompact(io, x)
-        textwidth(String(take!(io)))
+    function ourstrwidth(io::IO, x::Any) # -> Int
+        truncate(buffer, 0)
+        ourshow(IOContext(buffer, :compact=>get(io, :compact, true)), x)
```

Comment:
Isn't this redundant with what `ourshow` does?

Reply:
This is the most delicate part of this PR (and that is why I had to add `io` argument to `ourstringwidth`).

`ourstrwidth` prints to `buffer` not `io` so we have to make sure that `buffer` inherits `:compact` from `io` (this would produce a wrong result if `io` had `:compact` set to `false` as `buffer` by default does not have it defined, so `ourshow` would print to `buffer` assuming `:compact` is `true`).